src/Xtools/RFA.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/**
+ * An RFA object contains the parsed information for an RFA
+ *
+ * @category RFA
+ * @package  Xtools
+ * @author   Xtools Team <xtools@lists.wikimedia.org>
+ * @license  GPL 3.0
+ * @link     http://xtools.wmflabs.org/rfa
+ */
+
+namespace Xtools;
+
+/**
+ * Class RFA
+ *
+ * @category RFA
+ * @package  Xtools
+ * @author   Xtools Team <xtools@lists.wikimedia.org>
+ * @license  GPL 3.0
+ * @link     http://xtools.wmflabs.org/rfa
+ */
+class RFA
+{
+    private $sections;
+    private $data;
+    private $duplicates;
+    private $user_looking_for;
+    private $userSectionFound;
+    private $endDate;
+
+    /**
+     * Attempts to find a signature in $input using the default regex.
+     * Returns matches.
+     *
+     * @param string $input   The line we're looking for
+     * @param array  $matches Pointer to an array where we stash results
+     *
+     * @TODO: Make this cleaner
+     *
+     * @return int
+     */
+    protected function findSig($input, &$matches)
+    {
+        //Supports User: and User talk: wikilinks, {{fullurl}},
+        // unsubsted {{unsigned}}, unsubsted {{unsigned2}},
+        // anything that looks like a custom sig template
+        // TODO: Cross-wiki this sucker
+        $regexp
+            = //1: Normal [[User:XX]] and [[User talk:XX]]
+            "/\[\[[Uu]ser(?:[\s_][Tt]alk)?\:([^\]\|\/]*)(?:\|[^\]]*)?\]\]"
+            //2: {{fullurl}} and {{unsigned}} templates
+            . "|\{\{(?:[Ff]ullurl\:[Uu]ser(?:[\s_][Tt]alk)?\:|"
+            . "[Uu]nsigned\|)([^\}\|]*)(?:|[\|\}]*)?\}\}"
+            //3: {{User:XX/sig}} templates
+            . "|(?:\{\{)[Uu]ser(?:[\s_][Tt]alk)?\:([^\}\/\|]*)"
+            //4: {{unsigned2|Date|XX}} templates
+            . "|\{\{[Uu]nsigned2\|[^\|]*\|([^\}]*)\}\}"
+            //5: [[User:XX/sig]] links (compromise measure)
+            . "|(?:\[\[)[Uu]ser\:([^\]\/\|]*)\/[Ss]ig[\|\]]/";
+
+        return preg_match_all(
+            $regexp,
+            $input,
+            $matches,
+            PREG_OFFSET_CAPTURE
+        );
```

Comment:
Oh it's quite fragile.  Problem is, with custom signatures this is the least invasive way to do it.  I'd like to ditch the regex though, if you have any ideas I'd love to hear them!